mage_ai/data_cleaner/transformer_actions/column.py

```diff
@@ -126,7 +135,7 @@ def impute(df, action, **kwargs):
                 )
             sample = df.loc[valid_idx, column].sample(len(invalid_idx), replace=True)
             sample.index = invalid_idx
-            df.loc[invalid_idx, column] = sample
+            df.at[invalid_idx, column] = sample
```

Comment:
From my testing `loc` is very unstable when it comes to assigning lists to values. 

For example, consider testing with assigning to the third column on the row below:
<img width="1088" alt="image" src="https://user-images.githubusercontent.com/71106771/175183992-820feab4-6720-4e46-9cbc-c39591a5e5da.png">
The code below will correctly assign the list to the row:
<img width="495" alt="image" src="https://user-images.githubusercontent.com/71106771/175183733-82640742-8d9f-4941-a22a-14b9e04230f6.png">
However the code below will throw an error:
<img width="495" alt="image" src="https://user-images.githubusercontent.com/71106771/175183778-8209dd22-f073-4bea-bc50-5df3c760c426.png">

This is due to the fact that `loc` supports assignments with a list, which it interprets as assigning a set of values to a set of cells rather than assigning the set of a single cell multiple times (and the latter is necessary to fill in NA values).

`at` will always work since it only targets a single cell at a time.

Reply:
Honestly not sure why that works, my guess is that `at` performs a loop over all the (row, col) indices passed and does the assignment 1 by 1.

Reply:
The explicit way to avoid any unexpected behavior would be to manually loop through each sample and use `at` to fill in the value properly, then the batching behavior observed can be avoided. I can work on that.

Reply:
Before that I'll add the explicit Python loop just to avoid any future bugs and so `at` is used as intended. I saw no errors when testing but it seems like working with lists in DataFrames has many edge cases so I'd rather be safe.

Reply:
Okay should be good now